File: mobile-app/lib/models/propietario_model.dart
class Propietario {

  String id;

  String nombre;

  String documento;

  String telefono;

  String correo;

  bool sincronizado;

  Propietario({
    required this.id,
    required this.nombre,
    required this.documento,
    required this.telefono,
    required this.correo,
    required this.sincronizado,
  });

  Map<String, dynamic> toMap() {

    return {
      'id': id,
      'nombre': nombre,
      'documento': documento,
      'telefono': telefono,
      'correo': correo,
      'sincronizado': sincronizado ? 1 : 0
    };

  }

  factory Propietario.fromMap(Map<String, dynamic> map) {

    return Propietario(
      id: map['id'],
      nombre: map['nombre'],
      documento: map['documento'],
      telefono: map['telefono'],
      correo: map['correo'],
      sincronizado: map['sincronizado'] == 1,
    );

  }

}

File: backend/src/models/propietarioModel.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Propietario = sequelize.define('Propietario', {

  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },

  nombre: {
    type: DataTypes.STRING,
    allowNull: false
  },

  documento: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true
  },

  telefono: {
    type: DataTypes.STRING
  },

  correo: {
    type: DataTypes.STRING
  },

  fecha_registro: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW
  }

}, {
  tableName: 'propietarios',
  timestamps: false
});

module.exports = Propietario;